node_modules/gh-users/lib/api.js

```diff
@@ -295,6 +296,64 @@ var updateUser = module.exports.updateUser = function(ctx, userId, update, callb
 };
 
 /**
+ * Change a user's local password
+ *
+ * @param  {Context}        ctx                     Standard context containing the current user and the current app
+ * @param  {string}         userId                  The id of user for which to change the local password
+ * @param  {string}         newPassword             The new password for the user
+ * @param  {string}         [oldPassword]           The previous password for the user. This is only required when the current user is not an administrator
+ * @param  {Function}       callback                Standard callback function
+ * @param  {Object}         callback.err            An error that occurred, if any
+ */
+var changePassword = module.exports.changePassword = function(ctx, userId, newPassword, oldPassword, callback) {
+    userId = GrasshopperUtil.getNumberParam(userId);
+
+    var validator = new Validator();
+    validator.check(null, {'code': 401, 'msg': 'Only authenticated users can update a user'}).isLoggedInUser(ctx);
+    validator.check(userId, {'code': 400, 'msg': 'A valid user id must be provided'}).isInt();
+    validator.check(newPassword, {'code': 400, 'msg': 'A valid new password must be provided'}).notEmpty();
+    validator.check(newPassword, {'code': 400, 'msg': 'A valid new password must be provided'}).len(6);
+    if (validator.hasErrors()) {
+        return callback(validator.getFirstError());
+    }
+
+    // Ensure the user exists
+    UsersDAO.getUser(userId, function(err, user) {
+        if (err) {
+            return callback(err);
+        }
+
+        // Ensure the user is using local authentication
+        if (!user.password) {
+            return callback({'code': 400, 'msg': 'This user is not using local authentication'});
+
+        // Regular users have to provide an old password
+        } else if (!ctx.user.canAdmin(user.AppId)) {
+            validator.check(oldPassword, {'code': 400, 'msg': 'A valid old password must be provided'}).notEmpty();
+            validator.check(oldPassword, {'code': 400, 'msg': 'A valid old password must be provided'}).len(6);
+            if (validator.hasErrors()) {
+                return callback(validator.getFirstError());
+            }
+        }
+
+        // Ensure the current user can change the user's password
+        UsersAuthz.canChangePassword(ctx, user, oldPassword, function(err, canChangePassword) {
+            if (err) {
+                return callback(err);
+            } else if (!canChangePassword) {
+                log().warn({'id': userId, 'actor': ctx.user.id}, 'Unauthorized attempt at changing a user\'s password');
+                return callback({'code': 401, 'msg': 'You are not allowed to update this user\'s password'});
+            }
+
+            var update = {
+                'password': AuthUtil.hashPassword(newPassword)
+            };
+            UsersDAO.updateUser(user, update, callback);
```

Comment:
Should have an `info` log that a user's password has changed, including the user who changed it